haystack/nodes/retriever/_base_encoder.py

```diff
@@ -0,0 +1,102 @@
+import logging
```

Comment:
Hey is it all right if we keep this in the `_embedding_encoder.py` file? I think the refactoring of this type is out of scope for this PR.

Reply:
Yes, it was necessary to prevent a circular import. Just after the circular import issue, I refactored everything. 😢

Reply:
`ImportError: cannot import name '_BaseEmbeddingEncoder' from partially initialized module 'haystack.nodes.retriever._embedding_encoder' (most likely due to a circular import) `

_OpenAIEmbeddingEncoder base class is _BaseEmbeddingEncoder, OpenAI encoder is located in another file as it was being imported in a safe way. Now that the safe was removed, I can merge back. Would you want it, or keep like this to make maintenance easier?

Bloating a file with too many subclasses will make it harder to maintain, e.g.: when updating OpenAI and Cohere at same time, merge issues will arise. And it wasn't a need.